pkg/registry/file/dynamicpathdetector/analyze_endpoints.go

```diff
@@ -0,0 +1,128 @@
+package dynamicpathdetector
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	types "github.com/kubescape/storage/pkg/apis/softwarecomposition"
+)
+
+func AnalyzeEndpoints(endpoints *[]types.HTTPEndpoint, analyzer *PathAnalyzer) ([]types.HTTPEndpoint, error) {
+	var newEndpoints []types.HTTPEndpoint
+	MergeDuplicateEndpoints(endpoints)
+	for _, endpoint := range *endpoints {
+		AnalyzeURL(endpoint.Endpoint, analyzer)
```

Comment:
you don't use return value and error?

Reply:
In segment of the code no, because I would just skip errors. 
We can talk about it